pkg/vminetworkscontroller/vmi_controller_test.go

```diff
@@ -402,3 +536,99 @@ func decorateVMWithUID(uid string, vm *virtv1.VirtualMachine) *virtv1.VirtualMac
 	vm.UID = apitypes.UID(uid)
 	return vm
 }
+func dummyMarkedForDeletionVMI(nadName string) *virtv1.VirtualMachineInstance {
+	vmi := dummyVMI(dummyVMISpec(nadName))
+	vmi.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	vmi.ObjectMeta.Finalizers = []string{metav1.FinalizerDeleteDependents}
+
+	return vmi
+}
+
+func dummyMarkedForDeletionVMIWithActivePods(nadName string) *virtv1.VirtualMachineInstance {
+	vmi := dummyVMI(dummyVMISpec(nadName))
+	vmi.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	vmi.ObjectMeta.Finalizers = []string{metav1.FinalizerDeleteDependents}
+
+	vmi.Status.ActivePods = map[apitypes.UID]string{"podUID": "dummyNodeName"}
+	vmi.UID = apitypes.UID(dummyUID)
+
+	return vmi
+}
+
+func dummyMarkedForDeletionVM(nadName string) *virtv1.VirtualMachine {
+	vm := dummyVM(dummyVMISpec(nadName))
+	vm.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	vm.ObjectMeta.Finalizers = []string{metav1.FinalizerDeleteDependents}
+
+	return vm
+}
+
+var _ = Describe("shouldCleanFinalizers", func() {
+	DescribeTable("should determine if finalizers should be cleaned up",
```

Comment:
would it make sense to re-use the helpers you've introduced above in this test's entries ?

Reply:
i prefer not, in order to keep the objects as sparse and simple as possible

Reply:
that argument could be made about the helpers you've added ... Why have them ?...

But ok, we can move on.